src/main/java/io/zold/api/RtTransaction.java

```diff
@@ -79,12 +90,27 @@ public ZonedDateTime time() {
         throw new UnsupportedOperationException("time() not yet implemented");
     }
 
-    // @todo #15:30min Implement amount() by parsing the string representation
-    //  of transaction according to the pattern, described in the white
-    //  paper. Replace relevant test case with actual tests.
     @Override
-    public long amount() {
-        throw new UnsupportedOperationException("amount() not yet implemented");
+    public long amount() throws IOException {
+        final String amnt = new IoCheckedScalar<>(
+            () -> new ItemAt<>(
+                2, new SplitText(this.transaction, ";")
+            ).value().asString()
+        ).value();
+        if (!RtTransaction.AMT.matcher(amnt).matches()) {
+            throw new IOException(
+                new UncheckedText(
+                    new FormattedText(
+                        String.format(
```

Comment:
@carlosmiranda the idea is to use `FormattedText` to format the text, not to wrap an already formatted text.

Reply:
@victornoel duh, was in a hurry :))